alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultUserAccessor.java

```diff
@@ -70,49 +71,56 @@ public DefaultUserAccessor(UserRepository userRepository, UserRoleRepository use
     }
 
-    private UserModel createModel(UserEntity user) {
-        List<UserRoleRelation> roleRelations = userRoleRepository.findAllByUserId(user.getId());
-        List<Long> roleIdsForUser = roleRelations.stream().map(UserRoleRelation::getRoleId).collect(Collectors.toList());
-        Set<UserRoleModel> roles = authorizationUtility.getRoles(roleIdsForUser);
-        return UserModel.of(user.getUserName(), user.getPassword(), user.getEmailAddress(), roles);
-    }
-
     @Override
-    public UserModel addOrUpdateUser(UserModel user) {
-        return addOrUpdateUser(user, false);
+    public UserModel addUser(String userName, String password, String emailAddress) throws AlertDatabaseConstraintException {
+        return addUser(UserModel.newUser(userName, password, emailAddress, Collections.emptySet()), false);
     }
 
     @Override
-    public UserModel addOrUpdateUser(UserModel user, boolean passwordEncoded) {
-        String password = passwordEncoded ? user.getPassword() : defaultPasswordEncoder.encode(user.getPassword());
-        UserEntity userEntity = new UserEntity(user.getName(), password, user.getEmailAddress());
-
-        Optional<UserEntity> existingUser = userRepository.findByUserName(user.getName());
-        Long userId = null;
-        if (existingUser.isPresent()) {
-            userId = existingUser.get().getId();
-            userEntity.setId(userId);
+    public UserModel addUser(UserModel user, boolean passwordEncoded) throws AlertDatabaseConstraintException {
+        String username = user.getName();
+        Optional<UserEntity> userWithSameUsername = userRepository.findByUserName(username);
+        if (userWithSameUsername.isPresent()) {
+            throw new AlertDatabaseConstraintException(String.format("A user with username '%s' is already present", username));
         }
 
-        authorizationUtility.updateUserRoles(userId, user.getRoles());
+        String password = passwordEncoded ? user.getPassword() : defaultPasswordEncoder.encode(user.getPassword());
+        UserEntity newEntity = new UserEntity(username, password, user.getEmailAddress());
+        UserEntity savedEntity = userRepository.save(newEntity);
+        UserModel model = createModel(savedEntity);
+
+        authorizationUtility.updateUserRoles(model.getId(), user.getRoles());
 
-        return createModel(userRepository.save(userEntity));
+        return model;
     }
 
     @Override
-    public UserModel addUser(String userName, String password, String emailAddress) {
-        return addOrUpdateUser(UserModel.of(userName, password, emailAddress, Collections.emptySet()));
+    public UserModel updateUser(UserModel user, boolean passwordEncoded) throws AlertDatabaseConstraintException {
+        Long userId = user.getId();
+        if (null == userId) {
+            throw new AlertDatabaseConstraintException("A user id must be specified");
+        }
+
+        userRepository.findById(userId)
```

Comment:
Is this check only to verify that a user exists? We could use the query to check for existence instead

Reply:
It's to check that the id is valid before attempting to continue.